src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java

```diff
@@ -121,7 +121,7 @@
     private String jenkinsTunnel;
     @CheckForNull
     private String credentialsId;
-    private int containerCap = Integer.MAX_VALUE;
+    private Integer containerCap = Integer.MAX_VALUE;
```

Comment:
Prefer

```suggestion
    private Integer containerCap;
```

revert the patch to `readResolve` and change the getter to return

```java
containerCap != null ? containerCap ? Integer.MAX_VALUE
```

to avoid gratuitously serializing an ugly value from an unconfigured field. (You will anyway once someone `save`s the cloud, unless you also null out the default from the getter.)